### PR TITLE
Fyst 1026 implement id form 40 xml and pdf income

### PR DESCRIPTION
**Pull request by tahsinaislam**
## Link to pivotal/JIRA issue
https://codeforamerica.atlassian.net/browse/FYST-1026

## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

## What was done?
- Add calculations for ID 40 lines 7-11
- Added lines 7-11 to xml and pdf

## How to test?
- Unit tests were added for the ID40 calculator, xml and pdf filler

## Screenshots (for visual changes)
Screenshot 2024-11-07 at 5 07 21 PM

Screenshot 2024-11-07 at 5 06 44 PM

### CODE REVIEW

File: app/lib/efile/id/id_39r_calculator.rb [hunk 1]
```diff
@@ -1,29 +0,0 @@
...
-        [
-          @direct_file_data.total_qualifying_dependent_care_expenses,
-          [12_000 - @direct_file_data.excluded_benefits_amount, 0].max,
-          @direct_file_data.primary_earned_income_amount,
-          @direct_file_data.spouse_earned_income_amount,
-        ].min
-      end
-    end
-  end
-end
```
Comment (arinchoi03) [inline app/lib/efile/id/id_39r_calculator.rb:29]: why was this deleted?
 maybe you need to merge main? b/c i see commits made by me on this PR

File: app/lib/efile/id/id40_calculator.rb [hunk 2]
```diff
@@ -50,6 +55,26 @@ def calculate_line_6d
...
+      def calculate_line_9
+        line_or_zero(:ID40_LINE_7) + line_or_zero(:ID40_LINE_8)
+      end
+
+      def calculate_line_10
+        line_or_zero(:ID39R_B_LINE_24)
+      end
+
+      def calculate_line_11
+        line_or_zero(:ID40_LINE_9) - line_or_zero(:ID40_LINE_10)
```
Comment (arinchoi03) [inline app/lib/efile/id/id40_calculator.rb:75]: is it possible for this value to return a negative number? Is it ok to return a negative number?
I _think_ this will error if negative

https://www.w3.org/TR/xmlschema-2/#decimal

File: spec/lib/efile/id/id40_calculator_spec.rb [hunk 3]
```diff
@@ -67,6 +68,53 @@
...
+    end
+    it "returns federal AGI from direct file data" do
+      instance.calculate
+      expect(instance.lines[:ID40_LINE_7].value).to eq(12_501)
+    end
+  end
+
+  describe "Line 8: Additions from Form 39R" do
+    it "returns value from ID39R form line 7" do
+      allow(instance).to receive(:line_or_zero).and_call_original
```
Comment (arinchoi03) [inline spec/lib/efile/id/id40_calculator_spec.rb:83]: is this necessary?

**Review by arinchoi03 — APPROVED** [review]
with questions!

File: app/lib/efile/id/id_39r_calculator.rb (hunk 1 above)
Comment (tahsinaislam) [reply app/lib/efile/id/id_39r_calculator.rb:29]: This was an extra id39r calculator class that was left from a merge before. Kevin and I discussed this and should be deleted

File: spec/lib/efile/id/id40_calculator_spec.rb (hunk 3 above)
Comment (tahsinaislam) [reply spec/lib/efile/id/id40_calculator_spec.rb:83]: Probably not I'll delete this!

File: app/lib/efile/id/id40_calculator.rb (hunk 2 above)
Comment (tahsinaislam) [reply app/lib/efile/id/id40_calculator.rb:75]: Oh good call I'll fix this!

Comment (tahsinaislam) [reply app/lib/efile/id/id40_calculator.rb:75]: talked with margaret and negative numbers are accepted and i checked the xml with a negative number and it did not give any errors. 
Screenshot 2024-11-08 at 3 28 30 PM

Comment (arinchoi03) [reply app/lib/efile/id/id40_calculator.rb:75]: oh interesting!

Comment (arinchoi03) [reply app/lib/efile/id/id40_calculator.rb:75]: oh i should've read the thing i linked more closely:
>decimal has a lexical representation consisting of a finite-length sequence of decimal digits (#x30-#x39) separated by a period as a decimal indicator. **An optional leading sign is allowed. If the sign is omitted, "+" is assumed.** Leading and trailing zeroes are optional. If the fractional part is zero, the period and following zero(es) can be omitted. For example: -1.23, 12678967.543233, +100000.00, 210.
